### PR TITLE
[JENKINS-55787] Switch labels from entry to checkbox

### PR DESCRIPTION
**Pull request by jsoref**
See [JENKINS-55787](https://issues.jenkins-ci.org/browse/JENKINS-55787).

This is split from #3895 -- it's independent

### Proposed changelog entries

* Switch labels from entry to checkbox

### Submitter checklist

- [x] JIRA issue is well described
- [x] Changelog entry appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
      * Use the `Internal: ` prefix if the change has no user-visible impact (API, test frameworks, etc.)
- [x] Appropriate autotests or explanation to why this change has no tests
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### CODE REVIEW

**Comment by daniel-beck** [comment]
@oleg-nenashev Why the label?

**Comment by oleg-nenashev** [comment]
My bad. I believe I messed up something while editing label descriptions

File: test/src/test/java/hudson/model/ParametersTest.java
Comment (varyvol) [inline test/src/test/java/hudson/model/ParametersTest.java]: Why does it happen?

Comment (jsoref) [reply test/src/test/java/hudson/model/ParametersTest.java]: In the old style, you have something roughly like:

setting-name | setting-main
------------ | -------------
Label | [x]

In the new, it's closer to:

setting-name | setting-main
------------ | -------------
~ | [x] Label

**Comment by varyvol** [comment]
@jsoref does something change in the UI?

**Comment by jsoref** [comment]
@varyvol: yes, as shown in the issue description

**Comment by varyvol** [comment]
Did you have then the opportunity to run ATH tests?

**Review by oleg-nenashev — APPROVED** [review]
Weak +1. I have not checked all the forms, but it looks reasonable in all cases I see.

@varyvol ATH should not be affected if they query forms using the common API. I do not think there is a need to run it here

**Comment by oleg-nenashev** [comment]
@jenkinsci/core Few extra reviews will be appreciated

File: core/src/main/resources/hudson/model/StringParameterDefinition/config.jelly
Comment (dwnusbaum) [inline core/src/main/resources/hudson/model/StringParameterDefinition/config.jelly]: In https://github.com/jenkinsci/workflow-cps-global-lib-plugin/pull/62#discussion_r273305172, @jglick requested that the `field` remain on the `entry`, although I'm not entirely sure why. Have you tested to make sure that this particular change works fine with `field` on `checkbox` rather then `entry`?

Comment (jglick) [reply core/src/main/resources/hudson/model/StringParameterDefinition/config.jelly]: `field` should be on entry. In this case it may not matter now, so long as `help` is being specified explicitly (rather than simply being `help-trim.html` in the same directory), but it is better style.

Comment (jsoref) [reply core/src/main/resources/hudson/model/StringParameterDefinition/config.jelly]: So, `/help/parameter/*` files are used by at least `copyartifact-plugin` and `credentials-plugin`.

The trim file appears to only be used here, so I'm willing to move it:
https://github.com/search?q=org%3Ajenkinsci+%22help%2Fparameter%2Ftrim.html%22&type=Code

-- I'll look into cleaning this up... although it's a bit of an extended mess.

**Comment by jglick** [comment]
Of course plugins should not assume particular core resource paths exist.

**Comment by jsoref** [comment]
@jglick : https://issues.jenkins-ci.org/browse/JENKINS-57346 ~24 plugins would need to be cleaned up to properly address your concern. And in doing so, that will mean that they'll each essentially be forking content... I'm not 100% certain that's the right choice, but...

**Comment by oleg-nenashev** [comment]
Still needs more approvals from @jenkinsci/core

File: core/src/main/resources/hudson/model/StringParameterDefinition/config.jelly
Comment (jsoref) [reply core/src/main/resources/hudson/model/StringParameterDefinition/config.jelly]: Thanks @dwnusbaum / @jglick

**Comment by jsoref** [comment]
@dwnusbaum, @jglick: updated...

**Review by dwnusbaum — APPROVED** [review]
I don't see any obvious problems and the ATH seems to be fine with the changes, so a weak +1 from me.

**Comment by oleg-nenashev** [comment]
Probably we should wait for the feedback from @jglick before merging

**Comment by oleg-nenashev** [comment]
bump @jglick . I tentatively plan to merge it on Thursday if there is no negative feedback

File: core/src/main/resources/hudson/model/BooleanParameterDefinition/help-boolean-default_zh_CN.html
Comment (daniel-beck) [inline core/src/main/resources/hudson/model/BooleanParameterDefinition/help-boolean-default_zh_CN.html]: This file should be removed.

Comment (jsoref) [reply core/src/main/resources/hudson/model/BooleanParameterDefinition/help-boolean-default_zh_CN.html]: Oops, sorry...

File: core/src/main/resources/hudson/model/BooleanParameterValue/value.jelly
Comment (daniel-beck) [inline core/src/main/resources/hudson/model/BooleanParameterValue/value.jelly]: Reported as a bug: https://issues.jenkins-ci.org/browse/JENKINS-59388

**Comment by daniel-beck** [comment]
https://issues.jenkins-ci.org/browse/JENKINS-59388 reports this as a bug.

File: core/src/main/resources/hudson/model/BooleanParameterValue/value.jelly
Comment (daniel-beck) [inline core/src/main/resources/hudson/model/BooleanParameterValue/value.jelly:31]: Caused https://issues.jenkins.io/browse/JENKINS-72179

**Pull request merged**
